duckduckgo/iOS
# Add parameter to multiple Autofill pixels

Task/Issue URL: https://app.asana.com/0/0/1205431733474502/f
Tech Design URL:
CC:

**Description**:
Adding a temporary "default state" parameter to some autofill pixels ahead of Autofill becoming "on by default"



**Steps to test this PR**:
Note: the parameter will currently always be `off` since Autofill is currently off by default for all users. The logic to make this dynamic will be based on a phased rollout remote feature will be tackled in another PR.
Also, https://fill.dev/ is a useful site for testing
1. Enable Autofill in the Logins screen
2. Run through the various Autofill actions to trigger the pixels from this list. Confirm they all have this parameter: `["default_state": "off"]`

**Prompt to disable setting:**
To test this, start with a clean install and Autofill enabled. Visit 3 different websites that have a login option and attempt to login to each. For each Save Prompt choose "Not Now". After the 3rd "Not Now" you'll be presented with the alert that fires these pixels:
- [x] `m_autofill_logins_save_disable-prompt_shown`
- [x] `m_autofill_logins_save_disable-prompt_autofill-kept`
- [x] `m_autofill_logins_save_disable-prompt_autofill-disabled`

**Save login prompt:**
- [x] `m_autofill_logins_save_login_inline_displayed`
- [x] `m_autofill_logins_save_login_inline_confirmed`
- [x] `m_autofill_logins_save_login_inline_dismissed`

**Save password prompt:**
(https://news.ycombinator.com/login is a useful site to test triggering the password only save prompt - in the login form enter only a password and tap login to trigger)
- [x] `m_autofill_logins_save_password_inline_displayed`
- [x] `m_autofill_logins_save_password_inline_confirmed`
- [x] `m_autofill_logins_save_password_inline_dismissed`

**Fill login prompt:**
- [x] `m_autofill_logins_fill_login_inline_manual_displayed`
- [x] `m_autofill_logins_fill_login_inline_manual_confirmed`
- [x] `m_autofill_logins_fill_login_inline_manual_dismissed`
- [x] `m_autofill_logins_fill_login_inline_autoprompt_displayed`
- [x] `m_autofill_logins_fill_login_inline_autoprompt_confirmed`
- [x] `m_autofill_logins_autoprompt_dismissed`

**Opening and toggling autofill logins setting:**
- [x] `m_autofill_settings_opened` (Triggered by accessing Logins from Settings screen)
- [x] `m_nav_autofill_menu_item_pressed` (Triggered by accessing Logins from browsing menu)
- [x] `m_autofill_logins_settings_enabled`
- [x] `m_autofill_logins_settings_disabled`

**Password generation prompts:**
- [x] `m_autofill_logins_password_generation_prompt_displayed`
- [x] `m_autofill_logins_password_generation_prompt_confirmed`
- [x] `m_autofill_logins_password_generation_prompt_dismissed`


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
